src/proxy/topic/TopicProxy.ts

```diff
@@ -0,0 +1,53 @@
+import {BaseProxy} from "../BaseProxy";
+import {ITopic} from "../ITopic";
+import {MessageListener} from "../MessageListener";
+import {UUID} from "../../core";
+import {TopicAddMessageListenerCodec} from "../../codec/TopicAddMessageListenerCodec";
+import {TopicRemoveMessageListenerCodec} from "../../codec/TopicRemoveMessageListenerCodec";
+import {TopicPublishCodec} from "../../codec/TopicPublishCodec";
+import { ListenerMessageCodec} from "../../listener/ListenerMessageCodec";
+import {ClientMessage, ClientMessageHandler} from "../../protocol/ClientMessage";
+import {assertNotNull} from "../../util/Util";
+
+export class TopicProxy<E> extends BaseProxy implements ITopic<E> {
+
+    publish(message: E): Promise<void> {
+        assertNotNull(message);
+        const data = this.toData(message);
+        const request = TopicPublishCodec.encodeRequest(this.name, data);
+        const partitionId = request.getPartitionId();
+        return this.encodeInvokeOnPartition<void>(TopicAddMessageListenerCodec,  partitionId, TopicAddMessageListenerCodec.decodeResponse);
+    }
+
+    addMessageListener(listener: MessageListener<E>): string {
+        assertNotNull(listener);
+        const message = TopicAddMessageListenerCodec.encodeRequest(this.name, false);
+        const codec = this.createItemListener(this.name, false)
+        TopicAddMessageListenerCodec.handle(message, () => {});
+        //I am unable to fix this part. Will be trying tomorrow
+        return this.listenerService.registerListener(codec,  (message => {
+            TopicAddMessageListenerCodec.handle(message, ()=> {})
+        }));
```

Comment:
```suggestion
        return this.listenerService.registerListener(codec, listener);
```

It should be this way IMO

Reply:
`<html>TS2345: Argument of type 'MessageListener<E>' is not assignable to parameter of type 'ClientMessageHandler'.<br/>Types of parameters 'message' and 'message' are incompatible.<br/>Type 'ClientMessage' is missing the following properties from type 'Message<E>': messageObject, publisher, publishingTime` the reason I tried this way is this

Reply:
yeah, you need to convert the `ClientMessage` to a topic `Message`. `TopicItemHandler` does that in java. 

Instead of it, you can pass something like the following to `registerListener` method:
```js
(clientMessage: ClientMessage) => {
    TopicAddMessageListenerCodec.handle(clientMessage, (item: Data, publishTime: Long, uuid: UUID) => {
    // And then write the equivalent of the java code here:
    /*

            Member member = getContext().getClusterService().getMember(uuid);
            Message message = new DataAwareMessage(name, item, publishTime, member, getSerializationService());
            listener.onMessage(message);

    */
    });
} 
```